`import-error` inconsistency between `except ImportError` and `with contextlib.suppress(ImportError)`

### Bug description

There's an inconsistency with `import-error` between `except ImportError` and `with contextlib.suppress(ImportError)`.
With the former Pylint doesn't complain when it can't find the module/package, but with the latter it does complain.

This can easily be reproduced with the following code:
```python
# pylint: disable=missing-module-docstring
import contextlib

try:
    import bacon  # pylint: disable=unused-import
except ImportError:
    pass

with contextlib.suppress(ImportError):
    import eggs  # pylint: disable=unused-import
```

### Configuration

_No response_

### Command used

```shell
pylint toast.py
```


### Pylint output

```shell
************* Module toast
toast.py:10:4: E0401: Unable to import 'eggs' (import-error)
```


### Expected behavior

It's consistent.

### Pylint version

```shell
pylint 2.15.0-a0
astroid 2.12.2
Python 3.10.5 (main, Jul 27 2022, 01:21:11) [GCC 10.2.1 20210130 (Red Hat 10.2.1-11)]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/7270.false_positive
+++ b/doc/whatsnew/fragments/7270.false_positive
@@ -1,0 +1,3 @@
+``import-error`` now correctly checks for ``contextlib.suppress`` guards on import statements.
+
+Closes #7270

--- a/pylint/checkers/utils.py
+++ b/pylint/checkers/utils.py
@@ -1064,6 +1064,57 @@ def get_exception_handlers(
     return []
 
 
+def get_contextlib_with_statements(node: nodes.NodeNG) -> Iterator[nodes.With]:
+    """Get all contextlib.with statements in the ancestors of the given node."""
+    for with_node in node.node_ancestors():
+        if isinstance(with_node, nodes.With):
+            yield with_node
+
+
+def _suppresses_exception(
+    call: nodes.Call, exception: type[Exception] = Exception
+) -> bool:
+    """Check if the given node suppresses the given exception."""
+    for arg in call.args:
+        inferred = safe_infer(arg)
+        if isinstance(inferred, nodes.ClassDef):
+            if inferred.name == exception.__name__:
+                return True
+        elif isinstance(inferred, nodes.Tuple):
+            for elt in inferred.elts:
+                inferred_elt = safe_infer(elt)
+                if (
+                    isinstance(inferred_elt, nodes.ClassDef)
+                    and inferred_elt.name == exception.__name__
+                ):
+                    return True
+    return False
+
+
+def get_contextlib_suppressors(
+    node: nodes.NodeNG, exception: type[Exception] = Exception
+) -> Iterator[nodes.With]:
+    """Return the contextlib suppressors handling the exception.
+
+    Args:
+        node (nodes.NodeNG): A node that is potentially wrapped in a contextlib.suppress.
+        exception (builtin.Exception): exception or name of the exception.
+
+    Yields:
+        nodes.With: A with node that is suppressing the exception.
+    """
+    for with_node in get_contextlib_with_statements(node):
+        for item, _ in with_node.items:
+            if isinstance(item, nodes.Call):
+                inferred = safe_infer(item.func)
+                if (
+                    isinstance(inferred, nodes.ClassDef)
+                    and inferred.qname() == "contextlib.suppress"
+                ):
+                    if _suppresses_exception(item, exception):
+                        yield with_node
+
+
 def is_node_inside_try_except(node: nodes.Raise) -> bool:
     """Check if the node is directly under a Try/Except statement
     (but not under an ExceptHandler!).
@@ -1087,9 +1138,9 @@ def node_ignores_exception(
     excepts.
     """
     managing_handlers = get_exception_handlers(node, exception)
-    if not managing_handlers:
-        return False
-    return any(managing_handlers)
+    if managing_handlers:
+        return True
+    return any(get_contextlib_suppressors(node, exception))
 
 
 def class_is_abstract(node: nodes.ClassDef) -> bool:

--- a/pylint/checkers/utils.py
+++ b/pylint/checkers/utils.py
@@ -1045,7 +1045,7 @@ def _except_handlers_ignores_exceptions(
 
 
 def get_exception_handlers(
-    node: nodes.NodeNG, exception: type[Exception] = Exception
+    node: nodes.NodeNG, exception: type[Exception] | str = Exception
 ) -> list[nodes.ExceptHandler] | None:
     """Return the collections of handlers handling the exception in arguments.
 
@@ -1072,27 +1072,29 @@ def get_contextlib_with_statements(node: nodes.NodeNG) -> Iterator[nodes.With]:
 
 
 def _suppresses_exception(
-    call: nodes.Call, exception: type[Exception] = Exception
+    call: nodes.Call, exception: type[Exception] | str = Exception
 ) -> bool:
     """Check if the given node suppresses the given exception."""
+    if not isinstance(exception, str):
+        exception = exception.__name__
     for arg in call.args:
         inferred = safe_infer(arg)
         if isinstance(inferred, nodes.ClassDef):
-            if inferred.name == exception.__name__:
+            if inferred.name == exception:
                 return True
         elif isinstance(inferred, nodes.Tuple):
             for elt in inferred.elts:
                 inferred_elt = safe_infer(elt)
                 if (
                     isinstance(inferred_elt, nodes.ClassDef)
-                    and inferred_elt.name == exception.__name__
+                    and inferred_elt.name == exception
                 ):
                     return True
     return False
 
 
 def get_contextlib_suppressors(
-    node: nodes.NodeNG, exception: type[Exception] = Exception
+    node: nodes.NodeNG, exception: type[Exception] | str = Exception
 ) -> Iterator[nodes.With]:
     """Return the contextlib suppressors handling the exception.
 
@@ -1130,7 +1132,7 @@ def is_node_inside_try_except(node: nodes.Raise) -> bool:
 
 
 def node_ignores_exception(
-    node: nodes.NodeNG, exception: type[Exception] = Exception
+    node: nodes.NodeNG, exception: type[Exception] | str = Exception
 ) -> bool:
     """Check if the node is in a TryExcept which handles the given exception.
 

--- a/tests/functional/i/import_error.py
+++ b/tests/functional/i/import_error.py
@@ -79,3 +79,32 @@ import foo, bar # [multiple-imports]
 
 import foo
 import bar
+
+# Issues with contextlib.suppress reported in
+# https://github.com/PyCQA/pylint/issues/7270
+import contextlib
+with contextlib.suppress(ImportError):
+    import foo2
+
+with contextlib.suppress(ValueError):
+    import foo2  # [import-error]
+
+with contextlib.suppress(ImportError, ValueError):
+    import foo2
+
+with contextlib.suppress((ImportError, ValueError)):
+    import foo2
+
+with contextlib.suppress((ImportError,), (ValueError,)):
+    import foo2
+
+x = True
+with contextlib.suppress(ImportError):
+    if x:
+        import foo2
+    else:
+        pass
+
+with contextlib.suppress(ImportError):
+    with contextlib.suppress(TypeError):
+        import foo2

--- a/tests/functional/i/import_error.txt
+++ b/tests/functional/i/import_error.txt
@@ -3,3 +3,4 @@ import-error:21:4:21:26::Unable to import 'maybe_missing_2':UNDEFINED
 no-name-in-module:33:0:33:49::No name 'syntax_error' in module 'functional.s.syntax':UNDEFINED
 syntax-error:33:0:None:None::Cannot import 'functional.s.syntax.syntax_error' due to 'invalid syntax (<unknown>, line 1)':HIGH
 multiple-imports:78:0:78:15::Multiple imports on one line (foo, bar):UNDEFINED
+import-error:90:4:90:15::Unable to import 'foo2':UNDEFINED

